### PR TITLE
Feature/873 noxgenerators extend generator.nox.yaml

### LINKED ISSUE #873: [Nox.Generators] Extend generator.nox.yaml
- Extend the generator.nox.yaml by adding a new ui property
    - this property is true or false
- The following validation needs to be added if UI **is not** NULL all other properties needs to be false.
- this property will control code generation for UI projects

### PR DESCRIPTION
**Pull request by anna-naumova25**

### CODE REVIEW

File: src/Nox.Generator/Common/GeneratorConfig.cs [hunk 1]
```diff
@@ -9,5 +12,13 @@ internal class GeneratorConfig
         public bool Infrastructure { get; set; } = true;
 
         public bool Presentation { get; set; } = true;
+
+        public UiType Ui { get; set; } = UiType.None;
```
Comment (rochar) [inline src/Nox.Generator/Common/GeneratorConfig.cs]: @anna-naumova25  we need to update this for a bool value, sorry we can catch up on this today

**Review by rochar — DISMISSED** [review]

File: tests/Nox.Generator.Tests/files/yaml/ui/invalid.generator.nox.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿#
+# generator.nox.yaml
+#
+# This file configures which layers of code the Nox generator will generate in this project.
```
Comment (ILoveMyTortoise) [inline tests/Nox.Generator.Tests/files/yaml/ui/invalid.generator.nox.yaml:4]: maybe describe in comment why it's wrong?

File: tests/Nox.Generator.Tests/files/yaml/ui/invalid.generator.nox.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# 
+# Please mark this as a C# analyzer addition file by including the following in your .csproj
+#
+#  <ItemGroup>
+#    <AdditionalFiles Include="generator.nox.yaml" />
+#  </ItemGroup>
+
+# Genrates momain models, events and related types
```
Comment (ILoveMyTortoise) [inline tests/Nox.Generator.Tests/files/yaml/ui/invalid.generator.nox.yaml]: Generates domain *
Fix everywhere, please

File: tests/Nox.Generator.Tests/files/yaml/ui/invalid.generator.nox.yaml (hunk 2 above)
Comment (anna-naumova25) [reply tests/Nox.Generator.Tests/files/yaml/ui/invalid.generator.nox.yaml:4]: added the corresponding comment

File: tests/Nox.Generator.Tests/files/yaml/ui/invalid.generator.nox.yaml (hunk 3 above)
Comment (anna-naumova25) [reply tests/Nox.Generator.Tests/files/yaml/ui/invalid.generator.nox.yaml]: have updated it in all the generator.nox.yaml

**Review by rochar — APPROVED** [review]

**Review by ILoveMyTortoise — APPROVED** [review]

**Pull request merged**